### PR TITLE
feat: support insert decimal.

### PR DESCRIPTION
**Pull request by youngsofun**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

the first step of  `Decimal literal/encoding` in https://github.com/datafuselabs/databend/issues/2931

also fix the decimal display (tested in insert tests)

Closes #issue

### CODE REVIEW

File: tests/sqllogictests/suites/base/11_data_type/11_0006_data_type_decimal [hunk 1]
```diff
@@ -137,3 +137,30 @@ query I
...
+## insert
+
+statement ok
+drop table if exists t_decimal_4_2;
+
+statement ok
+create table t_decimal_4_2(a Decimal(4,2));
+
+statement ok
+insert into t_decimal_4_2 values (1.1), (0.1), (1.0), (1.1e1), (1.1e-1);
```
Comment (BohuTANG) [inline tests/sqllogictests/suites/base/11_data_type/11_0006_data_type_decimal:150]: We need more edge tests for decimal, include negative.

Comment (youngsofun) [reply tests/sqllogictests/suites/base/11_data_type/11_0006_data_type_decimal:150]: @BohuTANG  ok, currently more tests is in unit test (faster for debug/modify)

Comment (BohuTANG) [reply tests/sqllogictests/suites/base/11_data_type/11_0006_data_type_decimal:150]: logic test is high-level test.
For example, we can do it as:
1, what's the max decimal define we can do in Databend.
2, Give a defination, like the test we have what's the range of it.


If we have these, it's easy to write documents and understand the decimal type better.

File: src/query/expression/src/deserializations/decimal.rs [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+        _rows: usize,
+        _format: &FormatSettings,
+    ) -> Result<()> {
+        todo!()
+    }
+
+    fn de_json(&mut self, value: &serde_json::Value, _format: &FormatSettings) -> Result<()> {
+        match value {
+            serde_json::Value::Number(n) => {
+                if n.is_i64() {
```
Comment (sundy-li) [inline src/query/expression/src/deserializations/decimal.rs:101]: Seems it's wrong, we should care about the `precision` ? See `m_integer_to_decimal`

**Review by BohuTANG — APPROVED** [review]

File: src/query/expression/src/deserializations/decimal.rs (hunk 2 above)
Comment (youngsofun) [reply src/query/expression/src/deserializations/decimal.rs]: @sundy-li  https://github.com/datafuselabs/databend/pull/10167/files#diff-93bbc5f6f54e7d2224ea2ff14e7dcf663af5d93c401dab999039c144f3cbf8d9R238

**Review by sundy-li — APPROVED** [review]

File: src/query/expression/src/deserializations/decimal.rs (hunk 2 above)
Comment (sundy-li) [reply src/query/expression/src/deserializations/decimal.rs]: Ok, better to add a test to cover this bug.
